src/util/DbrHelper.sol

```diff
@@ -0,0 +1,335 @@
+    /// @param dolaAmount Amount of DOLA available to repay
+    /// @return repaidAmount Actual amount of DOLA repaid
+    function _repay(
+        Repay calldata repay,
+        uint256 dolaAmount
+    ) internal returns (uint256 repaidAmount) {
+        uint256 debt = IMarket(repay.market).debts(repay.to);
+        repaidAmount = (dolaAmount * repay.percentage) / denominator;
+        // If repaidAmount is higher than debt, use debt instead
+        if (repaidAmount > debt) {
+            repaidAmount = debt;
+        }
+        // Repay debt
+        IMarket(repay.market).repay(repay.to, repaidAmount);
+
+        emit RepayDebt(msg.sender, repay.market, repay.to, repaidAmount);
+    }
+
+    /// @notice Claim DBR
+    /// @return amount of DBR claimed
+    function _claimDBR() internal returns (uint amount) {
+        IINVEscrow escrow = _getEscrow();
+
+        uint256 dbrClaimable = escrow.claimable();
+        if (dbrClaimable == 0) revert NoDbrToClaim();
+
+        escrow.claimDBRTo(address(this));
+        amount = dbr.balanceOf(address(this));
+        if (dbrClaimable > amount || amount == 0)
+            revert ClaimedWrongAmount(amount, dbrClaimable);
+    }
+
+    /// @notice Get escrow for the user
+    /// @return escrow Escrow for the user
+    function _getEscrow() internal view returns (IINVEscrow escrow) {
+        escrow = IINVEscrow(address(invMarket.escrows(msg.sender)));
+        if (address(escrow) == address(0)) revert NoEscrow(msg.sender);
+    }
+
+    /// @notice Sell DBR for DOLA or INV
+    /// @param amountIn Amount of DBR to sell
+    /// @param minOut Minimum amount of DOLA or INV to receive
+    /// @param indexOut Index of the token to receive (0 for DOLA, 2 for INV)
+    /// @param receiver Address to receive DOLA or INV
+    /// @return amountOut Amount of DOLA or INV received
+    function _sellDbr(
+        uint amountIn,
+        uint minOut,
+        uint indexOut,
+        address receiver
+    ) internal returns (uint256 amountOut) {
+        amountOut = curvePool.exchange(
+            dbrIndex,
+            indexOut,
+            amountIn,
+            minOut,
+            false,
+            receiver
+        );
+        emit Sell(msg.sender, amountIn, amountOut, indexOut, receiver);
+    }
+
+    /// @notice Check inputs
+    /// @param params ClaimAndSell struct with parameters for claiming and selling
+    /// @param repay Repay struct with parameters for repaying debt
+    function _checkInputs(
+        ClaimAndSell calldata params,
+        Repay calldata repay
+    ) internal view {
+        if (
+            params.toDbr == address(0) &&
+            params.sellForDola + params.sellForInv != denominator
+        ) revert AddressZero(address(dbr));
+        if (params.toDola == address(0) && params.sellForDola > 0)
+            revert AddressZero(address(dola));
+        if (params.toInv == address(0) && params.sellForInv > 0)
+            revert AddressZero(address(inv));
```

Comment:
Consider renaming error to `ReceiverAddressZero`, to avoid confusion.

Reply:
Fixed in [f446755](https://github.com/InverseFinance/FiRM/pull/66/commits/f44675526e646aab388f30b936371beb701648ff)